BoxContentSDK/BoxContentSDK/Clients/BOXContentClient.m

```diff
@@ -495,7 +497,10 @@ - (void)prepareRequest:(BOXRequest *)request
         BOXSharedItemRequest *shareItemRequest = (BOXSharedItemRequest *)request;
         shareItemRequest.sharedLinkHeadersHelper = self.sharedLinksHeaderHelper;
     }
-    
+    if ([request conformsToProtocol:@protocol(BOXBackgroundRequestProtocol)]) {
```

Comment:
I think this should maybe check for `self.shouldBackgroundRequestsPerformImmediately` before even checking the request, and probably shouldn't override if the user has set `request. shouldPerformRequestImmediately` some other way.

Reply:
done